src/core/utils/types.ts

```diff
@@ -0,0 +1,7 @@
+export interface PersistentApiProperties {
```

Comment:
I'm not sure I like this location. just `core/types` should be fine. The mention of `utils/types` was for global utility types like `MaybePromise`